### PR TITLE
Remove old block style and other deprecations

### LINKED ISSUE #181: Remove old `AmplitudeVector` style
... everywhere it says "will be removed in `0.16.0`", i.e., the old block label stuff.

### PR DESCRIPTION
**Pull request by maxscheurer**
As announced, this removes the old way of defining/accessing matrix/vector blocks (i.e. vec['s'] and so on).
Closes #181.

### CODE REVIEW

File: adcc/ExcitedStates.py [hunk 1]
```diff
@@ -540,29 +540,29 @@ def excitations(self):
...
-for dep_property in deprecated:
-    new_key = deprecated[dep_property]
-
-    def deprecated_property(self, key=new_key, old_key=dep_property):
-        warnings.warn(f"Property '{old_key}' is deprecated "
-                      " and will be removed in version 0.16.0."
-                      f" Please use '{key}' instead.")
-        return getattr(self, key)
-    setattr(ExcitedStates, dep_property, property(deprecated_property))
+# deprecated = {
```
Comment (apapapostolou) [inline adcc/ExcitedStates.py:543]: Do we have to keep this?

**Review by apapapostolou — COMMENTED** [review]
In `functions.py`, the two functions `linear_combination` and `contract` were also announced to be removed in version 0.16.

File: adcc/ExcitedStates.py (hunk 1 above)
Comment (maxscheurer) [reply adcc/ExcitedStates.py:543]: Nope. Removed it.

**Comment by maxscheurer** [comment]
> In `functions.py`, the two functions `linear_combination` and `contract` were also announced to be removed in version 0.16. This is also the case for the `set_block` method of `OneParticleOperator`.

Right, I removed `contract` and `set_block`. `linear_combination` is tagged to be removed in `0.17`.

File: adcc/AdcMatrix.py [hunk 2]
```diff
@@ -165,8 +164,7 @@ def __init__(self, method, hf_or_mp, block_orders=None, intermediates=None,
                                       variant=variant)
                 for block, order in self.block_orders.items() if order is not None
             }
-            # TODO Rename to self.block in 0.16.0
```
Comment (apapapostolou) [inline adcc/AdcMatrix.py:168]: So we renamed it to `self.blocks` instead of `self.block`? It doesn't really matter. I just want to be sure.

Comment (maxscheurer) [reply adcc/AdcMatrix.py:168]: I think "blocks" is more natural for this dictionary?

**Comment by maxscheurer** [comment]
All comments are resolved. Feel free to approve & see if you can squash-merge the PR @apapapostolou.

**Review by apapapostolou — APPROVED** [review]

**Pull request merged**